### PR TITLE
Downloadmanager

### PR DESCRIPTION
**Pull request by rashiq**
It's happening 🔥

I'm slowly committing all the the code.

### CODE REVIEW

**Comment by EladKeyshawn** [comment]
🎉 🎉 🎉🎉 🎉 🎉

**Comment by julianharty** [comment]
@rashiq Well done and great to see you're working on the Android app again. I wondered:
- Are you planning to use the GPL v3 license for the entire app? If so, it'd be good to use it consistently in all the source files and state who owns the copyright (presumably you), rather than just having an anonymous copyright statement.
- In BookDataSource.java do you intend all the elements to be public? why?

Thanks

Julian

File: android/src/org/kiwix/kiwixmobile/library/entity/LibraryNetworkEntity.java [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+ */
+package org.kiwix.kiwixmobile.library.entity;
+
+import java.util.List;
+import org.simpleframework.xml.Attribute;
+import org.simpleframework.xml.ElementList;
+import org.simpleframework.xml.Root;
+
+@Root(name = "library")
+public class LibraryNetworkEntity {
```
Comment (julianharty) [inline android/src/org/kiwix/kiwixmobile/library/entity/LibraryNetworkEntity.java:27]: What's this class represent? it seems to be data including a URL, what's the connection or link to a network?

**Comment by rashiq** [comment]
Hey @julianharty! 
Great to hear from you again! :)
- I was under the impression that you have to include the GPL v3 header in all your source files - we kinda neglected that and I was thinking of adding it to all the files. However I'm not particularly fond of having a name in the header since I think that discourages contributions, and makes a file shout "this is mine: do not touch". Is it possible to just put in `Copyright 2016 Kiwix <contact@kiwix.org>` in the header?
- All the members in `BookDataSource.java` are annotated with their representative field name in the database. I'm currently making use of an annotation processing library called SquiDb. The `BookDataSource.java` class will not be used directly, but it will generate code that will have proper getter/setters and those will have the correct access levels.

File: android/src/org/kiwix/kiwixmobile/library/entity/LibraryNetworkEntity.java (hunk 1 above)
Comment (rashiq) [inline android/src/org/kiwix/kiwixmobile/library/entity/LibraryNetworkEntity.java:27]: It's a bean model of a the kiwix.org/library.xml. I have not yet committed the glue code that does the actual parsing - about to commit that rn :)

**Comment by kelson42** [comment]
What is the status on this? No move these last 3 weeks.

**Comment by rashiq** [comment]
Good news everyone! :)
I have a really stable download manager! 
The downloading part is stable, everything else is still being worked on. UI wise it's definitely not finished at all.

But you can already start downloading stuff!
Start the `LibraryActivity` through ADB, or put `startActivity(new Intent(this, LibraryActivity.class));` into the `onCreate()` of the `KiwixMobileActivity` class, if you want to test it out.

I will put the download option in the overflow menu soon as well, but right now that's the only option to try it out.

**Comment by EladKeyshawn** [comment]
@rashiq Great work man! :grinning:

**Pull request merged**